src/gitchangelog/gitchangelog.py

```diff
@@ -1246,43 +1254,56 @@ def kolibree_output(data, opts={}):
             re.DOTALL,
         )
 
-    def md_title(label, level=1):
+    def render_title(label: str, level: int = 1) -> str:
         return "#" * level + " " + label.strip() + "\n"
 
-    def render_version(version):
+    def render_version(version: dict) -> str:
         title = "%s (%s)" % (version["tag"], version["date"]) \
                 if version["tag"] else \
                 opts["unreleased_version_label"]
-        s = md_title(title, level=2)
+        s = render_title(title, level=2)
 
         sections = version["sections"]
-        nb_sections = len(sections)
-        for section in sections:
+        if len(sections) != 1:
+            die("There can be exactly one section for kolibree output")
 
-            section_label = section["label"] if section.get("label", None) \
-                            else "Other"
+        jira_sections = {
+            section: [] for section in JIRA_ISSUETYPE_TO_SECTION.values()
+        }
 
-            if not (section_label == "Other" and nb_sections == 1):
-                s += "\n" + md_title(section_label, level=3) + "\n"
+        for commit in sections[0]["commits"]:
+            section, entry = render_commit(commit)
+            section = JIRA_ISSUETYPE_TO_SECTION[section]
+            jira_sections[section].append(entry)
 
-            for commit in section["commits"]:
-                s += render_commit(commit)
+        for section, entries in jira_sections.items():
+            if not entries:  # TODO: Do we want to keep section nevertheless?
+                continue
+            s += "\n" + render_title(section, level=3) + "\n"
+            for entry in entries:
+                s += entry
         return s
 
-    def render_commit(commit, opts=opts):
+    def render_commit(commit: str) -> tuple:
+        """
+        Parse commit and return Jira section, well issue type and commit text.
```

Comment:
Probably a typo, not sure what you wanted to say.
```suggestion
        Parse commit and return Jira section mapped to issue type and commit text.
```

Reply:
:) If you are looking from the Jira side, than its an issue type, section here in code in our case. Will adjust

Reply:
Adjusted